drivers/serial/uart_stm32.c

```diff
@@ -1715,6 +1721,7 @@ static const struct uart_stm32_config uart_stm32_cfg_##index = {	\
 	.parity = DT_INST_ENUM_IDX_OR(index, parity, UART_CFG_PARITY_NONE),	\
 	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(index),			\
 	.single_wire = DT_INST_PROP_OR(index, single_wire, false), \
+	.tx_rx_swap = DT_INST_PROP_OR(index, tx_rx_swap, false),	\
```

Comment:
As the prop is boolean with default false, _OR stuff is not required.
```suggestion
	.tx_rx_swap = DT_INST_PROP(index, tx_rx_swap),	\
```
(we might also need to clean up some of the other props)

Reply:
I think in this case it is needed since the `tx-rx-swap` is only added to the `st,stm32-usart` binding and not the `st,stm32-uart` binding, but that may be an error on my part. I don't have a good understanding of the feature differences between UART/USART peripherals.